CMakeLists.txt

```diff
@@ -1,30 +1,33 @@
 cmake_minimum_required(VERSION 3.15)
 
-project(boost-asio-windows-sspi CXX)
+project(boost-windows-sspi CXX)
 
 set(CMAKE_CXX_STANDARD 11)
 
+enable_testing()
+
 if (WIN32)
   set(Boost_USE_STATIC_LIBS ON)
   set(Boost_USE_MULTITHREADED ON)
   set(Boost_USE_STATIC_RUNTIME OFF)
-endif()
 
-find_package(Boost REQUIRED COMPONENTS date_time regex)
+  find_package(Boost REQUIRED COMPONENTS date_time regex)
 
-add_executable(https_client https_client.cpp)
-target_link_libraries(https_client PRIVATE
-  Boost::headers
-  Boost::date_time
-  Boost::regex
-  )
+  add_library(boost-windows-sspi INTERFACE)
+  target_link_libraries(boost-windows-sspi INTERFACE
+    Boost::headers
+    Boost::date_time
```

Comment:
Are you sure linking `date_time` and `regex` is necessary? I thought they're optional transitive dependencies.

Reply:
To be honest, I can't really remember why they were necessary and what depended on them. I'll look into that.